### PR TITLE
Fixed: False positive for custom selectors in `selector-pseudo-class-no-unknown` #2141

### LINKED ISSUE #2141: False positive for custom selectors in `selector-pseudo-class-no-unknown`
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

False positive for custom selectors in `selector-pseudo-class-no-unknown`.

> Which rule, if any, is this issue related to?

`selector-pseudo-class-no-unknown`

> What CSS is needed to reproduce this issue?

e.g.

```css
:--heading { 
  margin-top: 0 
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "selector-pseudo-class-no-unknown": true
  }
}
```

> Which version of stylelint are you using?

`7.6.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

`postcss-cssnext`

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
1:1    Unexpected unknown pseudo-class selector ":-heading" (selector-pseudo-class-no-unknown)
```

No time fixed this today.

### PR DESCRIPTION
**Pull request by MoOx**
> Which issue, if any, is this issue related to?

#2141

> Is there anything in the PR that needs further explanation?

First commit proves it's failing. 2nd fix it.

Hope I fixed this the right way.

Closes #2141

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 -   Fixed: `comment-whitespace-inside` no longer complains about `/*!` comments with non-space whitespace (e.g. newlines) ([#2121](https://github.com/stylelint/stylelint/pull/2121))
 -   Fixed: `selector-max-compound-selectors` ignores SCSS nested declarations ([#2102](https://github.com/stylelint/stylelint/pull/2102))
 -   Fixed: regression causing `--stdin-filename` in CLI and `codeFilename` in Node API to error if a non-existent filename is provided ([#2128](https://github.com/stylelint/stylelint/pull/2128))
+-   Fixed: False positive for custom selectors in `selector-pseudo-class-no-unknown` ([#2141](https://github.com/stylelint/stylelint/pull/2141))
```
Comment (evilebottnawi) [inline CHANGELOG.md]: Don't add lines in changelog, this added after merge PR

File: src/rules/selector-pseudo-class-no-unknown/index.js [hunk 2]
```diff
@@ -47,6 +47,9 @@ export default function (actual, options) {
           // Ignore pseudo-elements
           if (value.slice(0, 2) === "::") { return }
 
+          // Ignore custom selectors
+          if (value.slice(0, 3) === ":--") { return }
```
Comment (evilebottnawi) [inline src/rules/selector-pseudo-class-no-unknown/index.js]: Best solution is write util `isCustomSelector.js` (also add tests) and use it

**Comment by evilebottnawi** [comment]
@MoOx thanks for helping :+1:

**Comment by MoOx** [comment]
PR rebased on master ;)

**Comment by davidtheclark** [comment]
👍

**Comment by jeddy3** [comment]
Added to changelog as:

> -   Fixed: `selector-pseudo-class-no-unknown` no longer reports false positives for custom selectors ([#2147](https://github.com/stylelint/stylelint/pull/2147)).

**Pull request merged**